src/org/rascalmpl/interpreter/result/ConcreteSyntaxResult.java

```diff
@@ -142,7 +142,16 @@ public <U extends IValue, V extends IValue> Result<U> fieldUpdate(String name, R
 		
 		return super.fieldUpdate(name, repl, store);
 	}
-	
+
+	private static boolean hasField(IConstructor prod, Name field) {
+		IList syms = ProductionAdapter.getSymbols(prod);
```

Comment:
Because a ConcreteSyntaxResult is always a Tree.appl:
* there are two builtin fields to consider, namely `prod` and `args` (see ConstructorResult, with the definition of `data Tree = appl(Production prod, list[Args] args)` in mind.
* and if there are "keyword fields" a.k.a. keyword parameters, then I think this function should also consider them.